nvflare/private/fed/client/client_app_runner.py

```diff
@@ -28,19 +29,26 @@
 
 
 class ClientAppRunner:
-    def __init__(self) -> None:
+    def __init__(self, time_out=60.0) -> None:
         self.command_agent = None
+        self.timeout = time_out
 
     def start_run(self, app_root, args, config_folder, federated_client, secure_train):
         client_runner = self.create_client_runner(app_root, args, config_folder, federated_client, secure_train)
+        federated_client.set_client_runner(client_runner)
+        start = time.time()
+        while federated_client.communicator.cell is None:
+            time.sleep(1.0)
+            if time.time() - start > self.timeout:
+                raise RuntimeError("No cell created for communicator. Failed to start the ClientAppRunner.")
         federated_client.status = ClientStatus.STARTED
         client_runner.run(app_root, args)
 
+        federated_client.communicator.cell.stop()
```

Comment:
Instead of accessing the instance variable hierarchically, should we define a function/method in FL client called `stop_cell` and call that here?

Reply:
refactored